File: _src/05/Chapter5.CSharp/Maps/PersonTypeMap.cs
using System.Data.Entity.ModelConfiguration;
using Chapter5.CSharp.Models;

namespace Chapter5.CSharp.Maps
{
public class PersonTypeMap : EntityTypeConfiguration<PersonType>
{
public PersonTypeMap()
{
    ToTable("TypeOfPerson");
    Property(p => p.TypeName)
            .HasMaxLength(30)
            .IsRequired();
    HasMany(pt => pt.Persons)
        .WithOptional(p => p.PersonType)
        .HasForeignKey(p => p.PersonTypeId)
        .WillCascadeOnDelete(false);
}
}
}

File: _src/05/Chapter5.VB/Maps/PersonTypeMap.vb
Imports System.Data.Entity.ModelConfiguration
Imports Chapter5.VB.Models

Namespace Maps
Public Class PersonTypeMap
    Inherits EntityTypeConfiguration(Of PersonType)
    Public Sub New()
        ToTable("TypeOfPerson")
        Me.Property(Function(p) p.TypeName) _
            .HasMaxLength(30) _
            .IsRequired()
        HasMany(Function(pt) pt.Persons) _
            .WithOptional(Function(p) p.PersonType) _
            .HasForeignKey(Function(pt) pt.PersonTypeId) _
            .WillCascadeOnDelete(False)
    End Sub
End Class
End NameSpace